android-components/components/browser/state/src/main/java/mozilla/components/browser/state/reducer/EngineStateReducer.kt

```diff
@@ -45,6 +45,7 @@ internal object EngineStateReducer {
         is EngineAction.OptimizedLoadUrlTriggeredAction -> {
             state
         }
+        is EngineAction.SaveToPdfErrorAction,
```

Comment:
Adding it to this list means that it could throw the error saying, "You need to add _EngineMiddleware_ to your BrowserStore." which isn't true for this action.

I think we ought to be using our own exception here instead:

```suggestion
        is EngineAction.SaveToPdfErrorAction -> {
            throw IllegalStateException("You need to add a middleware to handle this action in your BrowserStore. ($action)")
        }
```

Reply:
👍🏽